src/main/java/org/commcare/session/RemoteQuerySessionManager.java

```diff
@@ -171,14 +169,16 @@ public void refreshItemSetChoices(Hashtable<String, String> userAnswers) {
             for (Enumeration en = userInputDisplays.keys(); en.hasMoreElements(); ) {
                 String promptId = (String)en.nextElement();
                 QueryPrompt queryPrompt = userInputDisplays.get(promptId);
-                if (queryPrompt.getInput() != null && queryPrompt.getInput().contentEquals(INPUT_TYPE_SELECT1)) {
+                if (queryPrompt.isSelect()) {
                     String answer = userAnswers.get(promptId);
                     populateItemSetChoices(queryPrompt);
-                    Vector<SelectChoice> items = queryPrompt.getItemsetBinding().getChoices();
-                    if (!checkForValidSelectValue(items, answer)) {
-                        // if it's not a valid select value, blank it out
-                        userAnswers.put(promptId, "");
-                        dirty = true;
+                    String[] selectedChoices = extractSelectChoices(answer);
+                    for (String selectedChoice : selectedChoices) {
+                        if (!checkForValidSelectValue(queryPrompt.getItemsetBinding(), selectedChoice)) {
+                            // if it's not a valid select value, blank it out
+                            userAnswers.put(promptId, answer.replace(selectedChoice, ""));
```

Comment:
will this handle the delimiters in `answer` correctly? Won't it just replace the value: `a#,#b#,#c` -> `a#,##,#,c`?

Could be worth a test.

Reply:
great catch! Corrected the code here and added a test on Formplayer PR